pkg/streamingpromql/optimize/ast/util_test.go

```diff
@@ -26,6 +28,21 @@ func preprocessQuery(t *testing.T, expr parser.Expr) (parser.Expr, error) {
 	return promql.PreprocessExpr(expr, timestamp.Time(dummyTimeRange.StartT), timestamp.Time(dummyTimeRange.EndT), 0)
 }
 
+func getOutputFromASTOptimizationPassWithQueryPlan(t *testing.T, ctx context.Context, input string, createOptimizerFunc func(prometheus.Registerer) optimize.ASTOptimizationPass) (*prometheus.Registry, parser.Expr) {
```

Comment:
Might be nice to introduce a helper method that takes a optimisation pass as a parameter that calls this new method that can be used by the tests that don't care about metrics.

eg. the collapse constants tests can use the extra method, while the prune toggles tests can use `getOutputFromASTOptimizationPassWithQueryPlan`